eclair-core/src/main/scala/fr/acinq/eclair/wire/FailureMessage.scala

```diff
@@ -97,6 +96,7 @@ object FailureMessageCodecs {
     .typecase(18, ("expiry" | uint32).as[FinalIncorrectCltvExpiry])
     .typecase(19, ("amountMsat" | millisatoshi).as[FinalIncorrectHtlcAmount])
     .typecase(21, provide(ExpiryTooFar))
+    .typecase(PERM, sha256.as[InvalidOnionPayload])
```

Comment:
Since the only thing that was blocking this PR (from the spec point of view) was a decision for that specific error code, I'm currently simply using PERM (which is valid and very generic) which should unblock merging this PR.
We can add the final failure code once the spec stabilizes in another PR.

Reply:
I'm not a big fan of mapping PERM to InvalidOnionPayload bidirectionnally (even if this is temporary).
LND mentioned in the IRC meeting that they currently use InvalidOnionKey for that scenario: maybe we should start by doing that until the error code is correctly spec-ed?